conf/modules.config

```diff
@@ -262,62 +249,33 @@ process {
     }
 
     withName: KREPORT2KRONA_CENTRIFUGE {
-        publishDir = [
-            path: { "${params.outdir}/Taxonomy/${meta.classifier}/${meta.id}" },
-            mode: params.publish_dir_mode,
-            pattern: "*.txt",
-            enabled: false
-        ]
+        publishDir = [path: { "${params.outdir}/Taxonomy/${meta.classifier}/${meta.id}" }, mode: params.publish_dir_mode, pattern: "*.txt", enabled: false]
     }
 
     withName: KRONA_KTIMPORTTAXONOMY {
-        publishDir = [
-            path: { "${params.outdir}/Taxonomy/${meta.classifier}/${meta.id}" },
-            mode: params.publish_dir_mode,
-            pattern: "*.html"
-        ]
+        publishDir = [path: { "${params.outdir}/Taxonomy/${meta.classifier}/${meta.id}" }, mode: params.publish_dir_mode, pattern: "*.html"]
     }
 
-    //pattern: "*.{fa.gz,log}" //'pattern' didnt work, probably because the output is in a folder, solved with 'saveAs'
     withName: MEGAHIT {
-        ext.args   = params.megahit_options ?: ''
-        publishDir = [
-            path: { "${params.outdir}/Assembly" },
-            mode: params.publish_dir_mode,
-            saveAs: { filename ->
-                filename.equals('versions.yml')
-                    ? null
-                    : filename.indexOf('.contigs.fa.gz') > 0
-                        ? filename
-                        : filename.indexOf('.log') > 0 ? filename : null
-            }
-        ]
+        ext.args   = { params.megahit_options ? params.megahit_options + "-m ${task.memory.toBytes()}" : "-m ${task.memory.toBytes()}" }
+        ext.prefix = { "MEGAHIT-${meta.id}" }
+        publishDir = [path: { "${params.outdir}/Assembly/MEGAHIT" }, mode: params.publish_dir_mode, pattern: "*.{fa.gz,log}"]
     }
 
-    withName: SPADES {
-        ext.args   = params.spades_options ?: ''
-        publishDir = [
-            path: { "${params.outdir}/Assembly/SPAdes" },
-            mode: params.publish_dir_mode,
-            pattern: "*.{fasta.gz,gfa.gz,log}"
-        ]
+    withName: METASPADES {
+        ext.args   = params.spades_options ?: '--meta'
+        ext.prefix = { "SPAdes-${meta.id}" }
+        publishDir = [path: { "${params.outdir}/Assembly/SPAdes" }, mode: params.publish_dir_mode, pattern: "*.{fasta.gz,gfa.gz,fa.gz,log}"]
     }
 
-    withName: SPADESHYBRID {
-        ext.args   = params.spades_options ?: ''
-        publishDir = [
-            path: { "${params.outdir}/Assembly/SPAdesHybrid" },
-            mode: params.publish_dir_mode,
-            pattern: "*.{fasta.gz,gfa.gz,log}"
-        ]
+    withName: METASPADESHYBRID {
+        ext.args   = params.spades_options ?: '--meta'
```

Comment:
Yeah I wasn't sure what to do about that:

- I've already documented in the schema if you play with `--spades_options` you must include `--meta` (although I could imagine some people would want to run SPAdes alone)
- The other option is to just `+ '--meta'` to whatever is in `spades_options`, however I'm worried if someone includes `--meta` in their `--spades_options` the tool may complain of the duplicate...

I opted for version 1 to give more flexibility (but at the user's risk)

Reply:
I think if someone wants to use SPAdes without "--meta", nf-core/mag is likely the wrong pipeline, it should be rather nf-core/bacass. 
All options are not optimal I think. But forcing `--meta` is still more reliable than risking that people run SPAdes alone and get a very nice looking high depth assembly of one or two bins that is finally a heavily stitched chimera that represents 10% of all reads. I would prefer risking an error message with two `--meta` instead of risking people using SPAdes unknowingly without `--meta`. So `+ '--meta'` via the modules.config seems actually best to me. If someone really wants to remove that, doing that via a config is still possible.